### PR TITLE
Enable AutoExtern to consume from multiple source files

### PR DESCRIPTION
**Pull request by keyboardDrummer**
### Changes

Enable AutoExtern to consume from multiple source files

### Tests

Updated both AutoExtern tests to use this new feature

<small>By submitting this pull request, I confirm that my contribution is made under the terms of the [MIT license](https://github.com/dafny-lang/dafny/blob/master/LICENSE.txt).</small>

### CODE REVIEW

File: Source/AutoExtern/Program.cs [hunk 1]
```diff
@@ -398,10 +400,16 @@ public static class Program {
     return template;
   }
 
-  public static string GenerateDafnyCode(string projectPath, string filePath, string cSharpRootNS) {
-    var ast = AST.FromFile(projectPath, filePath, cSharpRootNS);
+  public static string GenerateDafnyCode(string projectPath, IList<string> sourceFiles, string cSharpRootNS) {
     var wr = new StringWriter();
-    ast.Pp(wr, "");
+    var asts = AST.FromFile(projectPath, sourceFiles, cSharpRootNS).ToList();
```
Comment (cpitclaudel) [inline Source/AutoExtern/Program.cs:405]:
```suggestion
    var asts = AST.FromFiles(projectPath, sourceFiles, cSharpRootNS).ToList();
```

File: Source/AutoExtern/Program.cs [hunk 2]
```diff
@@ -398,10 +400,16 @@ public static class Program {
...
+    var last = asts.Last();
+    foreach (var ast in asts) {
+      ast.Pp(wr, "");
+      if (ast != last) {
+        wr.WriteLine();
+      }
+    }
```
Comment (cpitclaudel) [inline Source/AutoExtern/Program.cs:412]:
```suggestion
    foreach (var ast in asts) {
      ast.Pp(wr, "");
      wr.WriteLine();
    }
```

File: Source/AutoExtern/Program.cs [hunk 3]
```diff
@@ -124,7 +124,7 @@ internal class AST : PrettyPrintable {
     this.model = model;
   }
 
-  public static AST FromFile(string projectPath, string filePath, string cSharpRootNS) {
+  public static IEnumerable<AST> FromFile(string projectPath, IEnumerable<string> sourceFiles, string cSharpRootNS) {
```
Comment (cpitclaudel) [inline Source/AutoExtern/Program.cs:127]:
```suggestion
  public static IEnumerable<AST> FromFiles(string projectPath, IEnumerable<string> sourceFiles, string cSharpRootNS) {
```

**Review by cpitclaudel — DISMISSED** [review]
LGTM, thanks.  Feel free to apply the suggestions or not.

File: Source/AutoExtern/Program.cs (hunk 2 above)
Comment (keyboardDrummer) [reply Source/AutoExtern/Program.cs:412]: Why this suggestion?

Comment (cpitclaudel) [reply Source/AutoExtern/Program.cs:412]: It simplifies the code and it doesn't cost anything.  Not a big deal.

**Review by cpitclaudel — APPROVED** [review]

File: Source/AutoExtern/Program.cs (hunk 2 above)
Comment (cpitclaudel) [reply Source/AutoExtern/Program.cs:412]: Also, it's marginally prettier, in that it doesn't depend on the sequence being non-empty? 🤷

Comment (keyboardDrummer) [reply Source/AutoExtern/Program.cs:412]: That's nice. I didn't want to have a double line in the output that's why I made this change.

Comment (cpitclaudel) [reply Source/AutoExtern/Program.cs:412]: OK, works for me :)

**Pull request merged**